File: C++/섹션06. DFS - 경우의 수 만들기/05-01. 조합 (combination), BOJ 15650 - N과 M (2).cpp
#include <stdio.h>

#define MAX (8 + 3)

int N, M;

int num_of_cases[MAX];

void printCases()
{
	for (int i = 0; i < M; i++) printf("%d ", num_of_cases[i]);
	putchar('\n');
}

void DFS(int depth, int start)
{
	if (depth == M)
	{
		printCases();
		return;
	}

	for (int i = start; i <= N; i++)
	{
		num_of_cases[depth] = i;

		DFS(depth + 1, i + 1);
	}
}

int main()
{
	scanf("%d %d", &N, &M);

	DFS(0, 1);

	return 0;
}

File: Python/섹션06. DFS - 경우의 수 만들기/05-01. 조합 (combination), BOJ 15650 - N과 M (2).py
MAX = 8 + 3

N, M = map(int, input().split())

num_of_cases = [0] * MAX

def print_cases():
    print(*num_of_cases[:M])

def DFS(depth, start):
    if depth == M:
        print_cases()
        return
    
    for i in range(start, N + 1):        
        num_of_cases[depth] = i
        
        DFS(depth + 1, i + 1)    
                 
DFS(0, 1)